src/go/rpk/pkg/cli/cluster/connections/connections.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2025 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+// Package connections deals with listing current connections in the cluster
+package connections
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	adminv2 "buf.build/gen/go/redpandadata/core/protocolbuffers/go/redpanda/core/admin/v2"
+	"connectrpc.com/connect"
+	"github.com/docker/go-units"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/adminapi"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+const defaultPageSize = 32
+
+var errInvalidStateFilter = errors.New("invalid state filter")
+
+var stateOptionMap = map[string]adminv2.KafkaConnectionState{
+	"OPEN":   adminv2.KafkaConnectionState_KAFKA_CONNECTION_STATE_OPEN,
+	"CLOSED": adminv2.KafkaConnectionState_KAFKA_CONNECTION_STATE_CLOSED,
+}
+
+var tableHeaders = []string{
+	"NODE",
+	"SHARD",
+	"UID",
+	"CLIENT-ID",
+	"IP:PORT",
+	"OPEN-TIME",
+	"IDLE",
+	"PROD-TPUT",
+	"FETCH-TPUT",
+	"REQS/MIN",
+	"STATE",
+}
+
+func NewCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "connections",
+		Short: "Manage and monitor cluster connections",
+		Long:  `Manage and monitor cluster connections.`,
+	}
+
+	cmd.AddCommand(newConnectionList(fs, p))
+	return cmd
+}
+
+func getConnectionDuration(conn *adminv2.KafkaConnection) string {
+	opened := conn.OpenTime.AsTime()
+	closed := conn.CloseTime.AsTime()
+
+	duration := closed.Sub(opened)
+	if opened.After(closed) {
+		duration = time.Since(opened)
+	}
+
+	return duration.Round(time.Second).String()
+}
+
+func writeConnectionRow(w *out.TabWriter, conn *adminv2.KafkaConnection) {
+	w.Print(
+		conn.NodeId,
+		conn.ShardId,
+		conn.Uid,
+		conn.ClientId,
+		fmt.Sprintf("%s:%d", conn.Source.IpAddress, conn.Source.Port),
+		getConnectionDuration(conn),
+		conn.IdleDuration.AsDuration().Round(time.Millisecond).String(),
+		units.HumanSize(float64(conn.RecentRequestStatistics.ProduceBytes/60)),
+		units.HumanSize(float64(conn.RecentRequestStatistics.FetchBytes/60)),
+		conn.RecentRequestStatistics.RequestCount,
+		strings.Replace(conn.State.String(), "KAFKA_CONNECTION_STATE_", "", 1),
```

Comment:
[nit - effectively the same but in case you want to use less arguments]
```suggestion
		strings.TrimPrefix(conn.State.String(), "KAFKA_CONNECTION_STATE_"),
```